Override Quantized Backend to use Fbgemm in Qlinear Packed Params Test

salilsdesai opened the pull request:
Summary: After D39934051, we must explicitly ```override_quantized_engine('fbgemm')``` for this test to work

Test Plan:
```
buck test //caffe2/test:ao -- TestQlinearPackedParams
```

Before:
```
Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/5629499663624574
    ✓ ListingSuccess: caffe2/test:ao : 72 tests discovered (32.830)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_qnnpack (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (25.085)
    ✗ Fail: caffe2/test:ao - test_qlinear_packed_params (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (26.706)
Test output:
> RuntimeError: Didn't find engine for operation ao::sparse::qlinear_prepack X86
```

After:
```
Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/7599824485968786
    ✓ ListingSuccess: caffe2/test:ao : 72 tests discovered (31.082)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_fbgemm (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (100.409)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_qnnpack (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (100.544)
Summary
  Pass: 2
  ListingSuccess: 1
```

Differential Revision: D40078176

jerryzh168 commented on test/ao/sparsity/test_qlinear_packed_params.py:
```diff
@@ -154,8 +154,14 @@ def test_qlinear_packed_params(self, allow_non_zero_zero_points=False):
                     self.assertEqual(y1, y2)
 
 
+    def test_qlinear_packed_params_fbgemm(self):
```
can we put @override_quantized_engine as a function decorator to reduce identation?

jerryzh168 replied:
oh sorry, I guess this is a contextmanager. I was talking about skipIfNoFBGEMM and skipIfNoQNNPACK, I guess we need to use both?

salilsdesai replied:
Good point. I'll add the skip decorators.

salilsdesai committed:
Override Quantized Backend to use Fbgemm in Qlinear Packed Params Test (#86236)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/86236

After D39934051, we must explicitly ```override_quantized_engine('fbgemm')``` for this test to work

Test Plan:
```
buck test //caffe2/test:ao -- TestQlinearPackedParams
```

Before:
```
Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/5629499663624574
    ✓ ListingSuccess: caffe2/test:ao : 72 tests discovered (32.830)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_qnnpack (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (25.085)
    ✗ Fail: caffe2/test:ao - test_qlinear_packed_params (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (26.706)
Test output:
> RuntimeError: Didn't find engine for operation ao::sparse::qlinear_prepack X86
```

After:
```
Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/7599824485968786
    ✓ ListingSuccess: caffe2/test:ao : 72 tests discovered (31.082)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_fbgemm (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (100.409)
    ✓ Pass: caffe2/test:ao - test_qlinear_packed_params_qnnpack (ao.sparsity.test_qlinear_packed_params.TestQlinearPackedParams) (100.544)
Summary
  Pass: 2
  ListingSuccess: 1
```

Reviewed By: jmdetloff

Differential Revision: D40078176

fbshipit-source-id: edb2852bcfc8476e1a79d64bb0000cd33f75e878
